### DISCUSSION TITLE
How to pass parameter to a create form ?

### DISCUSSION THREAD

**Discussion by MagicDidou** [filamentphp/filament#14660] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.81

### How can we help you?

Hi everybody,

Here is the thing. I have a table of `Supplier` with a row action redirecting to the creation of `SupplierInvoice` in its dedicated `CreateRecord`. I would like the creation form of the `SupplierInvoice` be able to use the related `Supplier` record (used to come here).

Here is the action code
```php
Tables\Actions\Action::make('Facture +')
   ->icon('heroicon-s-folder-plus')
   ->tooltip('Ajouter une facture')
   ->color('gray')
   ->url( fn($record):string=>SupplierInvoiceResource::getUrl('create', ['supplier_id' => $record->id])),
```

In the `CreateSupplierInvoice` class I have added a `supplier_id` member 
```php
    public $supplier_id = null ;
```
The URL is good, I reach the Creation Form and the parameter is present in the URL but I don't know how to retrieve this parameter in the Form. How can I use this id in the Form of the `SupplierInvoice` to automatically populate some fields ?

Any help will be appreciated

Did

**Comment by leandrocfe** [comment, marked as answer]
Maybe injecting [$request](https://filamentphp.com/docs/3.x/forms/advanced/#injecting-dependencies-from-laravels-container) ?

**Reply by tahinuralam** [reply]
Please make tutorial on it --FilamentPHP

this is a exam result form. after clicking on submit button it show the marks according to there roll number for every subjects that he gets in a new page and student can print there mark clicking on print button

Thanks in advance

http://www.educationboardresults.gov.bd/

**Comment by MagicDidou** [comment]
Great ! It works perfectly !

Thanks

**Reply by tahinuralam** [reply]
I have the same problem 

How to use the $request method 
I am new to filament

**Reply by MagicDidou** [reply]
Here is an example from my own project :
```php

                Forms\Components\Select::make('supplier_id')
                    ->relationship('supplier', 'name')
                    ->label('Supplier')
                    ->searchable()
                    ->preload()
                    ->required()
                    ->default( function(Request $request ) : string {
                        if ( $request->get('supplier_id')  === null )
                            return '';
                        return $request->get('supplier_id') ;
                    }),
                Forms\Components\DatePicker::make('deadline')
                    ->label('Deadline')
                    ->displayFormat('d/m/Y')
                    ->timezone('Europe/Paris')
                    ->default(function (Request $request): string {
                        return Supplier::getInvoiceDeadline( $request->get('supplier_id'))->format('Y-m-d');
                    }),
                Forms\Components\TextInput::make('description')
                    ->label('Description')
                    ->default(function (Request $request): string {
                        return Supplier::getNameFromId( $request->get('supplier_id')) ;
                    })
                    ->maxLength(255),
...
```

I populate the default using the data associated to a `Supplier` which ID is given as a request parameter.

**Be careful** : if you have components which generate requests (such as `Select` which searchable set to TRUE) and `live()` activated, then the function using the Request may be called several times.

Enjoy
